### DISCUSSION TITLE
How to Get a User's Location ?

### DISCUSSION THREAD

**Discussion by mhsfh** [tauri-apps/tauri#6048] [Q&A] [answered]
Hey there,
first of all: great tool! Thank you for developing this!
i'm working on desktop app and i need to get latitude, longitude of the user, i tried to use 
```js
navigator.geolocation.getCurrentPosition(successCallback, errorCallback);
```
but unfortunately it fail,

**Comment by tarekselem** [comment]
I have the same issue, and it doesn't show the permission requirement popup :(

**Comment by anandhuremanan** [comment]
If your front end framework is React.js you can get it by

```
const [lat, setLat] = useState<any>([]);
  const [long, setLong] = useState<any>([]);

  useEffect(() => {
    navigator.geolocation.getCurrentPosition(function (position) {
      setLat(position.coords.latitude);
      setLong(position.coords.longitude);
    });

    console.log("Latitude is:", lat);
    console.log("Longitude is:", long);
  }, [lat, long]);
```

It worked for me..

**Reply by tarekselem** [reply]
My implementation was exactly like that, but unfortunately didn't work with macOS production build mode.

**Comment by arashi-dev** [comment]
any solution?

**Comment by flozero** [comment]
This is how I achieved this. Update info.Plist on ios and similar for android. Not sure how yet but probably something like updating `AndroidManifest.xml` with <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />

for IOS:

```
<key>NSLocationWhenInUseUsageDescription</key>
<string>Your custom message explaining why you need access to the user's location.</string>
```

this will prompt the user to give access to geo when in use and you will have a second alert box in the window browser

I will probably create some plugins one day to get from rust binding

**Reply by Thor-x86** [reply]
For `AndroidManifest.xml`:

```xml
<!-- Always include this permission -->
<uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />

<!-- Include only if your app benefits from precise location access. -->
<uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />

<!-- Required only when requesting background location access on Android 10 (API level 29) and higher. -->
<uses-permission android:name="android.permission.ACCESS_BACKGROUND_LOCATION" />
```

Unfortunately, Android seems to not support native access for Location API according to [NDK Documentation][1].

[1]: https://developer.android.com/ndk/guides/stable_apis#android_native_application_apis

**Comment by acocalypso** [comment]
any update on that issue?
I'm not able to get lat and lon for any OS Build.

**Comment by laciferin2024** [comment]
Is this fixed?

**Reply by FabianLars** [reply, marked as answer]
For mobile there's https://github.com/tauri-apps/plugins-workspace/tree/v2/plugins/geolocation (the iOS and android prereqs in the guide should make the webview apis work as well i think). Desktop support is tracked here <https://github.com/tauri-apps/plugins-workspace/issues/2074> and that would/should then also polyfill the webview apis if needed.

**Comment by cellulosa** [comment]
I'd be interested too in getting this working on macos

**Comment by i16n** [comment]
Bumping this. Really need this to be a thing.
